glotaran/utils/io.py

```diff
@@ -252,3 +253,95 @@ def safe_dataframe_replace(
         to_be_replaced_values = [to_be_replaced_values]
     if column_name in df.columns:
         df[column_name].replace(to_be_replaced_values, replace_value, inplace=True)
+
+
+def extract_sas(
+    result: Result | xr.Dataset, species: str | None = None, dataset: str | None = None
+) -> xr.DataArray:
+    """Extract SAS data from a result.
+
+    This is a helper function to easily extract clp guidance spectra.
+
+    Parameters
+    ----------
+    result: Result | xr.Dataset
+        Optimization ``Result`` instance or result dataset.
+    species: str | None
+        Name op the species to extract the SAS for. Defaults to None
+    dataset: str | None
+        Name of the dataset to look up in a Result instance. Defaults to None
+
+    Returns
+    -------
+    xr.DataArray
+        SAS data with dummy time dimension.
+
+    Raises
+    ------
+    ValueError
+        If result is of type ``Result`` and ``dataset`` is None or not in the result data.
+    ValueError
+        If result is not of type ``Result`` or ``xr.Dataset``.
+    ValueError
+        If ``species`` is None or not in the species coordinates.
+    ValueError
+        If the result dataset does not contain a ``species_associated_spectra`` data variable.
+
+    Examples
+    --------
+    Extracting the SAS from an optimization result object.
+
+    .. code-block:: python
+
+        from glotaran.utils.io import extract_sas
+        from glotaran.io import save_dataset
+
+        sas = extract_sas(result, "species_1","dataset_1")
+        save_dataset("sas__result_dataset_1__species_1.ascii")
+
+
+    Extracting the SAS from a result dataset loaded from file.
+
+    .. code-block:: python
+
+        from glotaran.io import load_result
+        from glotaran.io import save_dataset
+        from glotaran.utils.io import extract_sas
+
+        result_dataset = load_result("result_dataset_1.nc")
+        sas = extract_sas(result_dataset, "species_1")
+        save_dataset("sas__result_dataset_1__species_1.ascii")
```

Comment:
see earlier comment `save_dataset(sas, 'filename.ascii)`

Reply:
Fixed in [90146e5](https://github.com/glotaran/pyglotaran/pull/1015/commits/90146e5d6c90e55e52795ed357f937e2ac1df19f)